go/vt/vtctl/grpcvtctldserver/server.go

```diff
@@ -628,12 +628,49 @@ func (s *VtctldServer) GetTablets(ctx context.Context, req *vtctldatapb.GetTable
 		}
 	}
 
-	if req.Keyspace != "" && req.Shard != "" {
-		tabletMap, err := s.ts.GetTabletMapForShard(ctx, req.Keyspace, req.Shard)
+	// Create a context for our per-cell RPCs, with a timeout upper-bounded at
+	// the RemoteOperationTimeout.
+	//
+	// Per-cell goroutines may also cancel this context if they fail and the
+	// request specified Strict=true to allow us to fail faster.
+	ctx, cancel := context.WithTimeout(ctx, *topo.RemoteOperationTimeout)
+	defer cancel()
+
+	var (
+		tabletMap map[string]*topo.TabletInfo
+		err       error
+	)
+
+	switch {
```

Comment:
Not sure to what extent this is a golang thing or a you thing, but I really like this (and other) uses of `switch` (instead of if/else, presumably). Going to osmose this pattern into our TypeScript more often. >:)

Reply:
Probably a mix of both haha